src/core/agent.py

```diff
@@ -35,10 +36,24 @@ def __init__(self, task: str, max_clarifications: int = 3, max_searches: int = 4
         self.conversation = []
         self.max_clarifications = max_clarifications
         self.max_searches = max_searches
-        self.openai_client = AsyncOpenAI(base_url=config.openai.base_url, api_key=config.openai.api_key)
+        # Initialize OpenAI client with optional proxy support
+        client_kwargs = {
+            "base_url": config.openai.base_url,
+            "api_key": config.openai.api_key
+        }
+        
+        # Add proxy if configured and not empty
+        if config.openai.proxy and config.openai.proxy.strip():
+            client_kwargs["http_client"] = self._create_http_client_with_proxy(config.openai.proxy)
```

Comment:
```suggestion
            client_kwargs["http_client"] = httpx.AsyncClient(proxy=proxy_url)
```

предложил бы сразу закидывать клиент чтобы не плодить функции

Reply:
Согласен, поправил